src/pretty/entry.rs

```diff
@@ -0,0 +1,117 @@
+                self.push(self.printer.convert_expr(expr));
+            } else if let Some(pattern) = node.cast() {
+                self.push_leading_space();
+                self.push(self.printer.convert_pattern(pattern));
+            } else if node.kind() == SyntaxKind::Colon {
+                self.push(BoxDoc::text(":"));
+            } else if node.kind() == SyntaxKind::Hash {
+                self.push_leading_space();
+                self.push(BoxDoc::text("#"));
+                self.leading = true;
+                continue;
+            } else if is_comment_node(node) {
+                self.push_leading_space();
+                self.push(comment(node));
+                if node.kind() == SyntaxKind::LineComment {
+                    self.push(BoxDoc::hardline());
+                    self.leading = true;
+                    continue;
+                }
+            } else {
+                // We ignore line break here.
+                continue;
+            }
+            self.leading = false;
+        }
+
+        self.doc.group()
+    }
+
+    pub fn convert_keyed(mut self, keyed: Keyed<'a>) -> BoxDoc<'a, ()> {
+        for node in keyed.to_untyped().children() {
+            if let Some(expr) = node.cast() {
+                self.push_leading_space();
+                self.push(self.printer.convert_expr(expr));
+            } else if node.kind() == SyntaxKind::Colon {
+                self.push(BoxDoc::text(":"));
+            } else if is_comment_node(node) {
+                self.push_leading_space();
+                self.push(comment(node));
+                if node.kind() == SyntaxKind::LineComment {
+                    self.push(BoxDoc::hardline());
+                    self.leading = true;
+                    continue;
+                }
+            } else {
+                // We ignore line break here.
+                continue;
+            }
+            self.leading = false;
+        }
+
+        self.doc.group()
+    }
+
+    pub fn convert_spread(mut self, spread: Spread<'a>) -> BoxDoc<'a, ()> {
+        for node in spread.to_untyped().children() {
+            if let Some(expr) = node.cast() {
+                self.push(self.printer.convert_expr(expr));
+            } else if node.kind() == SyntaxKind::Dots {
+                self.push(BoxDoc::text(".."));
+            } else if is_comment_node(node) {
+                self.push(comment(node));
+                if node.kind() == SyntaxKind::LineComment {
+                    self.push_leading_space();
+                    self.push(BoxDoc::hardline());
+                    self.leading = true;
+                    continue;
+                }
+            } else {
+                // We ignore line break here.
+                continue;
+            }
+            self.leading = false;
+        }
+
+        self.doc.group()
+    }
+
+    fn push(&mut self, doc: BoxDoc<'a, ()>) {
+        let a = std::mem::replace(&mut self.doc, BoxDoc::nil());
```

Comment:
this looks weird... What's this for?

Reply:
I havn't look into this in detail, but I guess we should simply use `Option<BoxDoc>` here.

Reply:
I want to simplify `self.doc = self.doc.append(...);`, but it has issues with move.
Maybe we should not be lazy here.